docs/openapi/definitions/auth.yaml

```diff
@@ -0,0 +1,217 @@
+            type: string
+            example: email
+        - name: code
+          in: query
+          description:
+            Short-term authorization code used to be exchanged for long-term
+            tokens
+          required: true
+          schema:
+            type: string
+        - name: state
+          in: query
+          description: Nonce value stored in state
+          required: true
+          schema:
+            type: string
+        - name: provider-nonce
+          in: cookie
+          description: Nonce value stored in cookie
+          required: true
+          schema:
+            type: string
+        - name: env
+          in: query
+          description: Parameter to indicate runtime environment
+          required: true
+          schema:
+            type: string
+            example: development
+      responses:
+        200:
+          description: Message received from consent popup
+          headers:
+            Set-Cookie:
+              description: Refresh cookie
+              schema:
+                $ref: '#/components/schemas/Refresh-Cookie'
+    post:
+      tags:
+        - provider
+      summary: TODO
+      responses:
+        default:
+          description: TODO
+  /refresh:
+    get:
+      tags:
+        - provider
+      summary: Handles refreshing of tokens due to reload or expiry
+      description:
+        If supported by the provider, the `/refresh` endpoint is responsible for
+        refreshing the tokens using a refresh-token stored in a cookie
+      parameters:
+        - name: scope
+          in: query
+          description: Scope requested for the auth session
+          required: true
+          schema:
+            type: string
+            example: email
+        - name: optional
+          in: query
+          description: Prevent the popup from being displayed
+          schema:
+            type: string
+        - name: x-requested-with
+          in: header
+          description: X-Requested-With header preventing CSRF
+          required: true
+          schema:
+            type: string
+            example: 'x-requested-with: XMLHttpRequest'
+        - name: provider-refresh-token
+          in: cookie
+          description: Refresh token used to refresh the current session
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: Successfully refreshed the session
+        401:
+          description: Validation error
+  /logout:
+    post:
+      tags:
+        - provider
+      summary: Logs user out of the current provider
+      description:
+        When logging out the current session is removed and if present the
+        cookie containing the refresh token is also removed
+      parameters:
+        - name: x-requested-with
+          in: header
+          description: X-Requested-With header preventing CSRF
+          required: true
+          schema:
+            type: string
```

Comment:
nit nit: add the example here as well? 

example: 'x-requested-with: XMLHttpRequest'

Reply:
I has to be set to that exact string right?